### PR TITLE
UBERF-7603: Support multiple transactors

### PR DESCRIPTION
**Pull request by haiodo**
UBERF-7603: Support multiple transactors

<sub><a href="https://front.hc.engineering/guest/platform?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2Njk3YjQwYjEwNmQ1ZmVkZWI1ZWI4YmUiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InBsYXRmb3JtIiwicHJvZHVjdElkIjoiIn0.G4A6B29woWzU8fOSQ80kQZGtpG5-nrf-cj6Iu7vNmRg">Huly&reg;: <b>UBERF-7608</b></a></sub>

### CODE REVIEW

File: server/account/src/operations.ts [hunk 1]
```diff
@@ -58,26 +59,64 @@ import { accountPlugin } from './plugin'
...
-  return transactor
+
+  const groups = groupByArray(endpoints.map(toTransactor), (it) => it.group)
+  let transactors = (groups.get(workspaceInfo.transactor ?? '') ?? [])
+    .map((it) => (kind === EndpointKind.Internal ? it.internalUrl : it.externalUrl))
+    .flat()
+
+  if (transactors.length === 0) {
+    // This is really bad
+    if (transactors.length === 0) {
```
Comment (BykhovDenis) [inline server/account/src/operations.ts]: Something wrong, duplicate previous line

File: .vscode/launch.json [hunk 2]
```diff
@@ -164,7 +163,7 @@
         "MINIO_SECRET_KEY": "minioadmin",
         "MINIO_ENDPOINT": "localhost",
         "MONGO_URL": "mongodb://localhost:27017",
-        "TRANSACTOR_URL": "ws://localhost:3333",
+        "ACCOUNTS_URL": "ws://localhost:3333",
```
Comment (BykhovDenis) [inline .vscode/launch.json]: ?

File: .vscode/launch.json [hunk 3]
```diff
@@ -124,7 +123,7 @@
       "request": "launch",
       "args": ["src/index.ts", "gen-recruit", "ws1", "20"],
       "env": {
-        "TRANSACTOR_URL": "ws://localhost:3333",
+        "ACCOUNTS_URL": "ws://localhost:3000",
```
Comment (BykhovDenis) [inline .vscode/launch.json:126]: Protocol?

File: .vscode/launch.json (hunk 2 above)
Comment (haiodo) [reply .vscode/launch.json]: Fixed.

File: .vscode/launch.json (hunk 3 above)
Comment (haiodo) [reply .vscode/launch.json:126]: Fixed.

File: server/account/src/operations.ts (hunk 1 above)
Comment (haiodo) [reply server/account/src/operations.ts]: Yep, fixed :-)

File: server/account/src/operations.ts [hunk 4]
```diff
@@ -58,26 +59,68 @@ import { accountPlugin } from './plugin'
...
   }
-  return endpoint
-}
+  const endpoints = transactorsUrl.split(',')
 
-const getTransactor = (): string => {
-  const transactor = getMetadata(toolPlugin.metadata.Transactor)
-  if (transactor === undefined) {
-    throw new Error('Please provide transactor url')
+  if (endpoints.length === 0) {
```
Comment (aonnikov) [inline server/account/src/operations.ts:115]: this condition is inaccurate
when the separator is not an empty string, then `split` never returns  an empty array. There will be at least one element, but the element might be an empty string. Probably need something like this:
```
const endpoints = transactorsUrl.split(',').filter((p) => p.length > 0)
```

**Review by aonnikov — DISMISSED** [review]

**Review by BykhovDenis — APPROVED** [review]

**Review by aonnikov — APPROVED** [review]

**Pull request merged**